### PR TITLE
Feature/connect tags to trends page

### PR DESCRIPTION
**Pull request by Barylo**
image
Now you can see tags on the right panel
And when you click on ShowMore link you will get to the next page with list of tags for you
image

### CODE REVIEW

File: src/components/AsyncSearch.tsx [hunk 1]
```diff
@@ -43,7 +43,13 @@ export const AsyncSearch = () => {
   );
 
   const handleSearch = (searchString: string) => {
-    dispatch(tagsActions.fetchTags({ searchString }));
+    dispatch(
+      tagsActions.fetchTags({
+        searchString,
+        limit: undefined,
+        nextPage: undefined,
```
Comment (alexandrtovmach) [inline src/components/AsyncSearch.tsx]: why do you need this?

File: src/components/AsyncSearch.tsx [hunk 2]
```diff
@@ -79,6 +85,7 @@ export const AsyncSearch = () => {
                   key={option._id}
                   _id={option._id}
                   name={option.name}
+                  stats={option.tweets}
```
Comment (alexandrtovmach) [inline src/components/AsyncSearch.tsx]: if I'm not wrong
```suggestion
                  stats={option.stats.tweets}
```

Comment (alexandrtovmach) [reply src/components/AsyncSearch.tsx]: replace:
```
typeof option === "object" ? (
// ...
```
with:
```
// eslint-disable-next-line @typescript-eslint/no-explicit-any
const isTag = (data: any): data is Tag => data && typeof data === "object";
// ...
isTag(option) ? (
  <TagItem
    key={option._id}
    _id={option._id}
    name={option.name}
    createdAt={option.createdAt}
    updatedAt={option.updatedAt}
  />
) : (
  <>{option}</>
)
```

**Review by alexandrtovmach — CHANGES_REQUESTED** [review]

**Comment by ArtemLigerko** [comment]
I think trends has something another functionality, it must shows all tags that often founds in tweets, not only tags what I search. Am I right?

File: src/redux/reducers/tags.ts [hunk 3]
```diff
@@ -21,19 +21,26 @@ const tagsInitialStore: TagsStore = {
...
-    params: {
-      query: { name: { $regex: searchString, $options: "i" } },
-    },
-  });
-  return response.data;
-});
+interface FetchTagsFunctionArgs {
+  limit: number | undefined;
+  nextPage: number | undefined;
+  searchString: string | undefined;
```
Comment (ArtemLigerko) [inline src/redux/reducers/tags.ts]:
```suggestion
  limit: number;
  nextPage: number;
  searchString: string;
```

**Comment by Barylo** [comment]
> I think trends has something another functionality, it must shows all tags that often founds in tweets, not only tags what I search. Am I right?

Yep, you are right. I'll find out how to make it.

File: src/components/AsyncSearch.tsx (hunk 1 above)
Comment (Barylo) [reply src/components/AsyncSearch.tsx]: This reducer was modified to be reused in two tasks.
The first purpose of it was only to fetch tags when you search for tags. It wasn't in use of render tags in Trends page. For this purpose, there was only "searchString" parameter. To use this reducer for the fetch and render tags at Trends page I added to it parameters of "limit" and "nextPage" which are not useful for the first purpose but is needed for the second purpose.
You can see this reducer at the previous stage in develop and here it's with mine additions.

**Comment by ArtemLigerko** [comment]
> > I think trends has something another functionality, it must shows all tags that often founds in tweets, not only tags what I search. Am I right?
> 
> Yep, you are right. I'll find out how to make it.

WM-Screenshots-20220925193315
..like this;

File: src/components/AsyncSearch.tsx (hunk 1 above)
Comment (ArtemLigerko) [reply src/components/AsyncSearch.tsx]: You can add additional actions for it

File: src/containers/BookmarksList.tsx [hunk 4]
```diff
@@ -111,7 +111,9 @@ const BookmarksList: React.FC = () => {
             )}
           />
         ) : (
-          t("bookmarksList.noBookmarks")
+          t(
+            "You have not any bookmarks yet. Please add some and you will see them here"
```
Comment (ArtemLigerko) [inline src/containers/BookmarksList.tsx]: use  i18next

File: src/containers/Trends.tsx [hunk 5]
```diff
@@ -13,37 +18,64 @@ const StyledLink = styled(LinkWithLanguageQueryParam)`
...
+  );
+
+  useEffect(() => {
+    dispatch(
+      tagsActions.fetchTags({
+        limit: tags.limit,
+        nextPage: tags.nextPage,
+        searchString: undefined,
+      })
+    );
```
Comment (ArtemLigerko) [inline src/containers/Trends.tsx]: you use this block two times. create arrow function and call it.

File: src/components/AsyncSearch.tsx (hunk 2 above)
Comment (Barylo) [reply src/components/AsyncSearch.tsx]: ts says it should be:
```
isTag(option) ? (
                <TagItem
                  key={option._id}
                  _id={option._id}
                  name={option.name}
                  createdAt={option.createdAt}
                  updatedAt={option.updatedAt}
                  stats={{
                    tweets: 0,
                  }}
                />
```

Comment (Barylo) [reply src/components/AsyncSearch.tsx]: > const isTag = (data: any): data is Tag => data && typeof data === "object";

with const isTag
ts do not allow  stats={option.stats.tweets}
ts says it should be  stats={option.stats}

without const isTag ts allow  stats={option.stats.tweets}

**Comment by Barylo** [comment]
> > > I think trends has something another functionality, it must shows all tags that often founds in tweets, not only tags what I search. Am I right?
> > 
> > 
> > Yep, you are right. I'll find out how to make it.
> 
> WM-Screenshots-20220925193315 ..like this;

When arter you first searched then you see those searched tags during current session, but when refresh it disapears. it should be remembered in storage what have been searched the other session before.

**Comment by ArtemLigerko** [comment]
> > > > I think trends has something another functionality, it must shows all tags that often founds in tweets, not only tags what I search. Am I right?
> > > 
> > > 
> > > Yep, you are right. I'll find out how to make it.
> > 
> > 
> 
> When arter you first searched then you see those searched tags during current session, but when refresh it disapears. it should be remembered in storage what have been searched the other session before.

You don`t needs search functionality for this, just make request and reply list of trends in "Trends for you" sitebar, use 'tags' reducer and add new actions for this

**Review by ArtemLigerko — APPROVED** [review]

**Review by DavidKorystov — APPROVED** [review]

File: src/components/TrendItem.tsx [hunk 6]
```diff
@@ -21,24 +21,24 @@ export const TrendItem = ({ trend }: { trend: MockTrend }) => {
       className="text-decoration-none  text-reset d-flex flex-row py-2"
     >
       <Container className="py-2 px-3">
-        <p className="pb-1 text-secondary">
-          {trend.categoryName} · {trend.categoryValue}
-        </p>
-        <p className="pb-1 fw-bold">{trend.contentRefName}</p>
+        <p className="pb-1 text-secondary">{trend.name}</p>
+        <p className="pb-1 fw-bold">{trend.name}</p>
```
Comment (KirillSerg) [inline src/components/TrendItem.tsx:25]: perhaps it is better not to duplicate the name, but to use the user.name

**Review by KirillSerg — APPROVED** [review]
lgtm

**Pull request merged**